spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/discovery/TtlSchedulerTests.java

```diff
@@ -16,76 +16,96 @@
-import org.springframework.cloud.consul.ConsulAutoConfiguration;
-import org.springframework.cloud.consul.support.ConsulHeartbeatAutoConfiguration;
-import org.springframework.cloud.consul.test.ConsulTestcontainers;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
+import org.springframework.beans.factory.support.StaticListableBeanFactory;
+import org.springframework.cloud.consul.serviceregistry.ApplicationStatusProvider;
+
+import static com.ecwid.consul.v1.health.model.Check.CheckStatus;
+import static com.ecwid.consul.v1.health.model.Check.CheckStatus.CRITICAL;
 import static com.ecwid.consul.v1.health.model.Check.CheckStatus.PASSING;
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static com.ecwid.consul.v1.health.model.Check.CheckStatus.WARNING;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 /**
+ * Unit tests for {@link TtlScheduler}.
+ *
  * @author Stéphane Leroy
+ * @author Chris Bono
  */
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = TtlSchedulerTests.TtlSchedulerTestConfig.class,
-		properties = { "spring.application.name=ttlScheduler",
-				"spring.cloud.consul.discovery.instance-id=ttlScheduler-id",
-				"spring.cloud.consul.discovery.heartbeat.enabled=true",
-				"spring.cloud.consul.discovery.heartbeat.ttlValue=2", "management.server.port=0" },
-		webEnvironment = RANDOM_PORT)
-@ContextConfiguration(initializers = ConsulTestcontainers.class)
-public class TtlSchedulerTests {
-
-	@Autowired
-	private ConsulClient consul;
+@ExtendWith(MockitoExtension.class)
+class TtlSchedulerTests {
+
+	@Mock
+	private HeartbeatProperties configuration;
+
+	@Mock
+	private ConsulClient client;
+
+	@Mock
+	private ApplicationStatusProvider applicationStatusProvider;
+
+	private TtlScheduler ttlScheduler;
+
+	@BeforeEach
+	void setup() {
+		StaticListableBeanFactory beanFactory = new StaticListableBeanFactory(
+				Collections.singletonMap("applicationStatusProvider", applicationStatusProvider));
+		ttlScheduler = new TtlScheduler(configuration, client,
+				beanFactory.getBeanProvider(ApplicationStatusProvider.class));
+		when(configuration.computeHeartbeatInterval()).thenReturn(Duration.ofMillis(2000));
+	}
 
 	@Test
-	public void should_send_a_check_before_ttl_for_all_services() throws InterruptedException {
```

Comment:
This is covered in[ConsulAutoServiceRegistrationTtlCheckTests](https://github.com/spring-cloud/spring-cloud-consul/pull/690/files#diff-426803e0ee5a103b6b2158053cf34c1c5770f0a93c882e8e9d234ed2b186f3d3R72)